tests/test_metrics.py

```diff
@@ -0,0 +1,145 @@
+            # two elements different (absolute difference of 1 and 4 respectively)
+            # float weights
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([2, 2, 3, 8]),
+                np.array([0.5, 0.5, 1, 1]),
+                1.5,
+            ),
+        ],
+    )
+    def test_gower_dist_numerical_old(self, a_num, b_num, weights, expected_metric):
+        metric = um.gower_dist_numerical_old(a_num, b_num, weights)
+
+        assert type(metric) == float
+        assert metric >= 0
+        assert metric == expected_metric
+
+    @pytest.mark.parametrize(
+        "a_num, b_num, weights, min_vals, max_vals, expected_metric",
+        [  # equal arrays
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([1, 2, 3, 4]),
+                np.array([1, 1, 1, 1]),
+                np.array([1, 2, 3, 4]),
+                np.array([1, 2, 3, 4]),
+                0,
+            ),
+            # one element different (absolute difference of 1)
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([2, 2, 3, 4]),
+                np.array([1, 1, 1, 1]),
+                np.array([1, 2, 3, 4]),
+                np.array([2, 2, 3, 4]),
+                4,
+            ),
+            # two elements different (absolute difference of 1 for each)
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([2, 1, 3, 4]),
+                np.array([1, 1, 1, 1]),
+                np.array([1, 1, 3, 4]),
+                np.array([1, 1, 3, 4]),
+                2,
+            ),
+            # equal arrays - double weight for the first element
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([1, 2, 3, 4]),
+                np.array([2, 1, 1, 1]),
+                np.array([1, 2, 3, 4]),
+                np.array([1, 2, 3, 4]),
+                0,
+            ),
+            # one element different (absolute difference of 1) but double weight
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([2, 2, 3, 4]),
+                np.array([2, 1, 1, 1]),
+                np.array([1, 2, 3, 4]),
+                np.array([1, 2, 3, 4]),
+                2,
+            ),
+            # one element different (absolute difference of 2) but double weight
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([3, 2, 3, 4]),
+                np.array([2, 1, 1, 1]),
+                np.array([1, 2, 3, 4]),
+                np.array([3, 2, 3, 4]),
+                5,
+            ),
+            # two elements different (absolute difference of 1 and 4 respectively)
+            # float weights
+            (
+                np.array([1, 2, 3, 4]),
+                np.array([2, 2, 3, 8]),
+                np.array([0.5, 0.5, 1, 1]),
+                np.array([1, 2, 3, 4]),
+                np.array([2, 2, 3, 4]),
+                3,
+            ),
+        ],
```

Comment:
The arguments  `min_vals, max_vals` should reflect the range of data. Meanly `( min_vals[0], max_vals[0] )` should be the range of values of the point first component analyzed (in a 3D space, it could represent the range of x_axis).
In UmapExperiment class, this range is calculated on training data and the same value is used for test data too (where we may have some out of scale values).
These 2 args are used for rescaling the components (because some feature value range may be very different from the others). So my proposal would be to add few cases like:
```
# One point with out of scale value
(
    np.array([1, 2, 3, 4]),
    np.array([100, 2, 3, 4]),
    np.array([2, 1, 1, 1]),
    np.array([1, 2, 3, 4]),
    np.array([2, 2, 3, 4])
),
# One point with high value, close to the upper bound of the value range
(
    np.array([1, 2, 3, 4]),
    np.array([100, 2, 3, 4]),
    np.array([2, 1, 1, 1]),
    np.array([100, 2, 3, 4]),
    np.array([2, 2, 3, 4])
),
# One point with low first component value and a high value of the value range upper bound
(
    np.array([1, 2, 3, 4]),
    np.array([2, 2, 3, 4]),
    np.array([2, 1, 1, 1]),
    np.array([100, 2, 3, 4]),
    np.array([2, 2, 3, 4])
)
```